### DISCUSSION TITLE
meta.links property in paginated response should be optional

### DISCUSSION THREAD

**Discussion by plamendp** [laravel/framework#35593] [Ideas]
Currently (v8.x), the JSON response, when LengthAwarePaginator is used is like this:

`
{
    "data": [],
    "links": { ... }
    "meta": {
        "current_page": 1,
        ...
        "links": [
            {
                "url": ...,
                "label": ...,
                "active": ...
            }
        ]
        ...
    }
}
`

While I understand the purpose of meta.links property, sometimes it might be very long (e.g. hundreds of pages available). Or maybe someone just don't need them.

I can't find an easy and elegant way to remove them from the response, except overwriting ResourceCollection method toResponse() to modify the final JSON response.

Would be nice if this property is optional. 

Maybe there should be some JsonResource->without() method, similar to with().

**Comment by xippios** [comment]
I found the following solution:

https://stackoverflow.com/questions/64044392/laravel-8-api-pagination-and-resource-links-obj-into-meta-key

Works well and should be stable for now.
